lib/realtime/broadcast_changes/postgres_replication.ex

```diff
@@ -0,0 +1,263 @@
+  * `output_plugin` - The output plugin to use. Default is `pgoutput`.
+  * `proto_version` - The protocol version to use. Default is `1`.
+  * `handler_module` - The module that will handle the data received from the replication stream.
+  * `metadata` - The metadata to pass to the handler module.
+
+  """
+  use Postgrex.ReplicationConnection
+  require Logger
+
+  defmodule Handler do
+    @type t :: module()
+    @doc """
+    The `call/2` callback is called by the `PostgresReplication` module to send messages to the parent process. It also sends back to the server connection a message in return if the user wants to.
+
+    ## Parameters
+      * `message` - The message to be sent to the parent process.
+      * `metadata` - The metadata to be passed to the handler module.
+
+    ## Returns
+      * `{:reply, [term]}` - The message to be sent to server connection. Read more in Realtime.Adapters.Postgres.Protocol
+      * `:noreply` - No message is sent back to the server.
+    """
+    @callback call(any, map()) :: {:reply, [term]} | :noreply
+  end
+
+  @derive {Inspect, except: [:connection_opts]}
+
+  @default_opts [
+    auto_reconnect: false,
+    sync_connect: true
+  ]
+  @type t :: %__MODULE__{
+          connection_opts: Keyword.t(),
+          table: String.t(),
+          schema: String.t(),
+          opts: Keyword.t(),
+          step:
+            :disconnected
+            | :check_replication_slot
+            | :create_publication
+            | :check_publication
+            | :create_slot
+            | :start_replication_slot
+            | :streaming,
+          publication_name: String.t(),
+          replication_slot_name: String.t(),
+          output_plugin: String.t(),
+          proto_version: integer(),
+          handler_module: Handler.t(),
+          metadata: map()
+        }
+  defstruct connection_opts: nil,
+            table: nil,
+            schema: "public",
+            opts: [],
+            step: :disconnected,
+            publication_name: nil,
+            replication_slot_name: nil,
+            output_plugin: "pgoutput",
+            proto_version: 1,
+            handler_module: nil,
+            metadata: %{}
+
+  def start_link(%__MODULE__{opts: opts, connection_opts: connection_opts} = attrs) do
+    opts = @default_opts |> Keyword.merge(opts) |> Keyword.merge(connection_opts)
+
+    case Postgrex.ReplicationConnection.start_link(__MODULE__, attrs, opts) do
+      {:ok, pid} -> {:ok, pid}
+      {:error, {:already_started, pid}} -> {:ok, pid}
+      {:error, {:bad_return_from_init, {:stop, reason, _}}} -> {:error, reason}
+      {:error, {:case_clause, {:disconnect, reason, _}}} -> {:error, reason}
+      {:error, error} -> {:disconnect, error}
+    end
+  end
+
+  @impl true
+  def init(attrs) do
+    Logger.info(
+      "Initializing connection with the status: #{inspect(attrs |> Map.from_struct() |> Map.drop([:connection_opts]))}"
+    )
```

Comment:
btw, since the Inspect implementation already drops :connection_opts, I think you can use it here too:

```suggestion
  def init(%__MODULE__{} = attrs) do
    Logger.info(
      "Initializing connection with the status: #{inspect(attrs, pretty: true}"
    )
```

Reply:
nice call! will change 👍